doc/configuration.md

```diff
@@ -0,0 +1,75 @@
+# Configuration
+
+## Files
+
+### `$BASH_COMPLETION_USER_FILE`
+
+Sourced late by `bash_completion`, pretty much after everything else.
+Use this file for example to load additional completions, and to remove
+and override ones installed by bash_completion. Defaults to
+`~/.bash_completion` if unset or null.
+
+### `$XDG_CONFIG_HOME/bash_completion`
+
+Sourced by the `bash_completion.sh` `profile.d` script. This file is
+suitable for definitions of all `COMP*` environment variables below.
+If `$XDG_CONFIG_HOME` is unset or null,`~/.config` is
+used instead of it.
+
+## Environment variables
```

Comment:
Maybe this should be processed in a separate PR, but I feel this should be "Shell variables" because these variables do not have to have the export attribute.

Reply:
Oh! I've never thought that there would be a difference. Changed.